README.md

```diff
@@ -0,0 +1,417 @@
+- A specific file can be specified on the command line:
+    `--config <path-to-file>`
+- A run-in-place build will look for the configuration file in
+    `location_of_exe/../minetest.conf` and also `location_of_exe/../../minetest.conf`
+
+Command-line options:
+---------------------
+- Use --help
+
+Compiling
+---------
+
+### Compiling on GNU/Linux
+
+Install dependencies. Here's an example for Debian/Ubuntu:
+
+    $ sudo apt-get install build-essential libirrlicht-dev cmake libbz2-dev libpng-dev libjpeg-dev libxxf86vm-dev libgl1-mesa-dev libsqlite3-dev libogg-dev libvorbis-dev libopenal-dev libcurl4-gnutls-dev libfreetype6-dev zlib1g-dev libgmp-dev libjsoncpp-dev
+
+For Fedora users:
+
+    $ sudo dnf install make automake gcc gcc-c++ kernel-devel cmake libcurl* openal* libvorbis* libXxf86vm-devel libogg-devel freetype-devel mesa-libGL-devel zlib-devel jsoncpp-devel irrlicht-devel bzip2-libs gmp-devel sqlite-devel luajit-devel leveldb-devel ncurses-devel doxygen spatialindex-devel bzip2-devel
+
+You can install git for easily keeping your copy up to date.
+If you don’t want git, read below on how to get the source without git.  
+This is an example for installing git on Debian/Ubuntu:
+
+    $ sudo apt-get install git
+
+For Fedora users:
+
+    $ sudo dnf install git
+
+Download source (this is the URL to the latest of source repository, which might not work at all times) using git:
+
+    $ git clone --depth 1 https://github.com/minetest/minetest.git
+    $ cd minetest
+
+Download minetest_game (otherwise only the "Minimal development test" game is available) using git:
+
+    $ git clone --depth 1 https://github.com/minetest/minetest_game.git games/minetest_game
+
+Download source, without using git:
+
+    $ wget https://github.com/minetest/minetest/archive/master.tar.gz
+    $ tar xf master.tar.gz
+    $ cd minetest-master
+
+Download minetest_game, without using git:
+
+    $ cd games/
+    $ wget https://github.com/minetest/minetest_game/archive/master.tar.gz
+    $ tar xf master.tar.gz
+    $ mv minetest_game-master minetest_game
+    $ cd ..
+
+Build a version that runs directly from the source directory:
+
+    $ cmake . -DRUN_IN_PLACE=TRUE
+    $ make -j <number of processors>
+
+Run it:
+
+    $ ./bin/minetest
+
+- Use cmake . -LH to see all CMake options and their current state
+- If you want to install it system-wide (or are making a distribution package),
+  you will want to use -DRUN_IN_PLACE=FALSE
+- You can build a bare server by specifying -DBUILD_SERVER=TRUE
+- You can disable the client build by specifying -DBUILD_CLIENT=FALSE
+- You can select between Release and Debug build by -DCMAKE_BUILD_TYPE=<Debug or Release>
+  - Debug build is slower, but gives much more useful output in a debugger
+- If you build a bare server, you don't need to have Irrlicht installed.
+  In that case use -DIRRLICHT_SOURCE_DIR=/the/irrlicht/source
+
+### CMake options
+
+General options:
+
+* BUILD_CLIENT        - Build Minetest client
```

Comment:
I don't like how the indents are lost in the Markdown projection. How about putting these params into a code block or table?